network.cpp:53:listen_on_port(): [ERROR] bind(): Cannot assign requested address

ProxySQL version 1.4.9 in ubuntu16.04

I have completed the configuration, but something wrong to limit the IP address and the listening port is not up.

------following is the log------
2018-07-23 17:03:03 [INFO] Exiting...
2018-07-23 17:03:03 [INFO] Shutdown angel process
2018-07-23 17:03:03 [INFO] ProxySQL version 1.4.9-percona-1.1
2018-07-23 17:03:03 [INFO] Detected OS: Linux AL19 4.4.0-85-generic #108-Ubuntu SMP Mon Jul 3 17:23:59 UTC 2017 x86_64
2018-07-23 17:03:03 [INFO] Starting ProxySQL
2018-07-23 17:03:03 [INFO] Sucessfully started
2018-07-23 17:03:03 [INFO] Angel process started ProxySQL process 28452
Standard ProxySQL Cluster rev. 0.1.0702 -- ProxySQL_Cluster.cpp -- Mon Jul 16 07:26:24 2018
Standard ProxySQL Statistics rev. 1.4.1027 -- ProxySQL_Statistics.cpp -- Mon Jul 16 07:26:24 2018
Standard ProxySQL HTTP Server Handler rev. 1.4.1031 -- ProxySQL_HTTP_Server.cpp -- Mon Jul 16 07:26:24 2018
Standard ProxySQL Admin rev. 0.2.0902 -- ProxySQL_Admin.cpp -- Mon Jul 16 07:26:24 2018
Standard MySQL Threads Handler rev. 0.2.0902 -- MySQL_Thread.cpp -- Mon Jul 16 07:26:24 2018
Standard MySQL Authentication rev. 0.2.0902 -- MySQL_Authentication.cpp -- Mon Jul 16 07:26:24 2018
2018-07-23 17:03:03 [INFO] Dumping current MySQL Servers structures for hostgroup ALL
2018-07-23 17:03:03 [INFO] Dumping mysql_servers
+--------------+----------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+-------------+
| hostgroup_id | hostname | port | weight | status | compression | max_connections | max_replication_lag | use_ssl | max_latency_ms | comment | mem_pointer |
+--------------+----------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+-------------+
+--------------+----------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+-------------+
2018-07-23 17:03:03 [INFO] Dumping mysql_servers_incoming
+--------------+----------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+
| hostgroup_id | hostname | port | weight | status | compression | max_connections | max_replication_lag | use_ssl | max_latency_ms | comment |
+--------------+----------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+
+--------------+----------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+
2018-07-23 17:03:03 [INFO] New mysql_replication_hostgroups table
2018-07-23 17:03:03 [INFO] New mysql_group_replication_hostgroups table
2018-07-23 17:03:03 [INFO] Dumping current MySQL Servers structures for hostgroup ALL
2018-07-23 17:03:03 [INFO] Dumping mysql_servers
+--------------+----------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+-------------+
| hostgroup_id | hostname | port | weight | status | compression | max_connections | max_replication_lag | use_ssl | max_latency_ms | comment | mem_pointer |
+--------------+----------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+-------------+
+--------------+----------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+-------------+
Standard Query Processor rev. 0.2.0902 -- Query_Processor.cpp -- Mon Jul 16 07:26:24 2018
In memory Standard Query Cache (SQC) rev. 1.2.0905 -- Query_Cache.cpp -- Mon Jul 16 07:26:24 2018
2018-07-23 17:03:03 network.cpp:53:listen_on_port(): [ERROR] bind(): Cannot assign requested address
2018-07-23 17:03:03 network.cpp:53:listen_on_port(): [ERROR] bind(): Cannot assign requested address
2018-07-23 17:03:03 network.cpp:53:listen_on_port(): [ERROR] bind(): Cannot assign requested address
2018-07-23 17:03:03 network.cpp:53:listen_on_port(): [ERROR] bind(): Cannot assign requested address
Standard MySQL Monitor (StdMyMon) rev. 1.2.0723 -- MySQL_Monitor.cpp -- Mon Jul 16 07:26:24 2018



 netstat -npl|egrep "6032|6033"
tcp        0      0 0.0.0.0:6032            0.0.0.0:*               LISTEN      28452/proxysql


i try to change the IP address to 0.0.0.0:6033, it be able to start, but 172.19.207.12:6033 will not do so.
thx a lot

--- a/lib/network.cpp
+++ b/lib/network.cpp
@@ -51,6 +51,13 @@ int listen_on_port(char *ip, uint16_t port, int backlog, bool reuseport) {
                 if (bind(sd, next->ai_addr, next->ai_addrlen) == -1) {
                         //if (errno != EADDRINUSE) {
                                 proxy_error("bind(): %s\n", strerror(errno));
+                                // in case of 'EADDRNOTAVAIL' suggest a solution to user. See #1614.
+                                if (errno == EADDRNOTAVAIL) {
+                                    proxy_info(
+                                       "Trying to 'bind()' failed due to 'EADDRNOTAVAIL'. If trying to bind to a "
+                                       "non-local IP address, make sure 'net.ipv4.ip_nonlocal_bind' is set to '1'\n"
+                                    );
+                                }
                                 close(sd);
                                 freeaddrinfo(ai);
                                 return -1;

